### PR TITLE
Tests: Replace complex tox dependency factors with "current" and "future" factors

### PR DESCRIPTION
**Pull request by willbarton**
This is a proposal to replace our version-specific factors for tox environments with a simpler `current` and `future` nomenclature.

This means to run the unittests on the current production version of Python, Django, and Wagtail, one would use:

```
tox -e unittest-current
```

To run them against upcoming versions of Python, Django, and Wagtail:

```
tox -e unittest-future
```

I've also added a convenience `tox -e unittest` (🎩 @higs4281) that runs identically to `unittest-current`.

I've removed the `fast` and `slow` factors here and environments here; everything will run "slow," i.e. with migrations. As part of this, I've also removed the `NoMigrations` `MIGRATION_MODULE` and its dependencies. The acceptance test env will now run with full migrations (again, 🎩 @higs4281).

The intent here is to simplify the name of our test environments and make the naming version-independent.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: docs/python-unit-tests.md [hunk 1]
```diff
@@ -60,33 +60,29 @@ You can select specific environments using `-e`.
 Running `tox` by itself is the same as running:
 
 ```sh
-tox -e lint -e unittest-py36-dj111-wag113-slow
+tox -e lint -e unittest-current
```
Comment (higs4281) [inline docs/python-unit-tests.md]: Could we make this environment plain "unittest" and assume it's current?

Comment (willbarton) [reply docs/python-unit-tests.md]: Absolutely, It'd just involve creating a new `testenv` that uses the `unittest-current` configuration. [That's what `lint` is](https://github.com/cfpb/cfgov-refresh/pull/5581/files#diff-b91f3d5bd63fcd17221b267e851608e8R160-R167)

Comment (willbarton) [reply docs/python-unit-tests.md]: Done!

File: tox.ini [hunk 2]
```diff
@@ -121,43 +108,27 @@ commands=
...
-commands=
-    {env:TRAVIS_CHECK_MIGRATIONS:echo "Skipping Travis-only step"}
+[future]
+# Configuration necessary for upcoming versions of Python, Django, and 
+# Wagtail
+# Note: This is not an env will not run if invoked.
+basepython=python3.6
+deps=
+    Django>=2.2,<2.3
+    wagtail>=2.3,<2.4
```
Comment (higs4281) [inline tox.ini]: Is it worth noting here that this section will need to be manually changed when we move the chains (unlike the 'current' configs)?

Comment (willbarton) [reply tox.ini]: Yes, it probably is.

File: tox.ini [hunk 3]
```diff
@@ -121,43 +108,27 @@ commands=
...
-#   tox -e unittest-py{36}-dj{111}-wag{23}-slow
-#
-# To run unit tests.
-setenv=
-    DJANGO_SETTINGS_MODULE=cfgov.settings.test
-
+[current]
+# Configuration necessary for the production versions Python, Django and 
+# Wagtail
+# Note: This is not an env will not run if invoked.
```
Comment (chosak) [inline tox.ini]: Could you rephrase this comment (and below)? I'm not entirely sure what this means.

Comment (willbarton) [reply tox.ini]: Certainly! I've also renamed all of these config sections to `-config` to ensure that the fact that they're configuration values rather than testenvs is clearer.

File: tox.ini (hunk 2 above)
Comment (willbarton) [reply tox.ini]: I've taken a stab at this in 3f28e5e. What do you think?

**Comment by higs4281** [comment]
If we drop the fast/slow routine, we should be able to drop this file:

- cfgov/cfgov/settings/test_nomigrations.py

... and these code blocks:

- https://github.com/cfpb/cfgov-refresh/blob/91beb6e620372a21e3475a25420ef21e7fc86b34/cfgov/core/utils.py#L174-L186
- https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/test.py#L68-L111

**Comment by chosak** [comment]
> If we drop the fast/slow routine...

@higs4281 that's a great idea, but I think we'll need to keep [this call](https://github.com/cfpb/cfgov-refresh/blob/ed714a7d7f553aac8e308dbcd9bcca6a8d917b24/cfgov/cfgov/test.py#L76-L78) to the initial data script, in some form.

**Comment by higs4281** [comment]
> > If we drop the fast/slow routine...
> 
> @higs4281 that's a great idea, but we might need to keep [this call](https://github.com/cfpb/cfgov-refresh/blob/ed714a7d7f553aac8e308dbcd9bcca6a8d917b24/cfgov/cfgov/test.py#L76-L78) to the initial data script, in some form.

Ah, thank you!

File: tox.ini (hunk 3 above)
Comment (willbarton) [reply tox.ini]: And I've fixed to missing "and", which is critical to that sentence.

**Comment by willbarton** [comment]
@higs4281 @chosak right now our [acceptance test settings use the no-migrations settings](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/settings/test_acceptance.py#L1) (and thus the [no-migrations `MIGRATIONS_MODULES`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/settings/test_nomigrations.py#L6)). 

Should that just run with migrations, or should we keep `NoMigrations` for that?

**Comment by willbarton** [comment]
I've pushed a change to remove all the no-migrations options and for the acceptance test settings to run migrations.

**Comment by chosak** [comment]
> Should that just run with migrations, or should we keep NoMigrations for that?

I suppose if we think that running with migrations always is fast enough, then it's better to be consistent than to introduce a difference in how the acceptance tests work.

**Comment by Scotchester** [comment]
The acceptance tests will soon have no option to run them in a way that involves Python.
